src/sema/ast.rs

```diff
@@ -875,6 +875,22 @@ impl Contract {
             })
             .cloned()
     }
+
+    pub fn functions_with_duplicate_names(&self, ns: &Namespace) -> Vec<usize> {
+        let mut map = HashMap::new();
+        let mut duplicates = Vec::new();
+
+        for func_no in &self.functions {
+            match map.insert(ns.functions[*func_no].id.name.clone(), func_no) {
+                None => (),
+                Some(prev) => {
+                    duplicates.push(*prev);
+                    duplicates.push(*func_no);
```

Comment:
Surely you'll end up with duplicates in the duplicates. Needs a test case if you really want to use this as-is